src/main/java/seedu/address/logic/commands/PrescribeCommand.java

```diff
@@ -0,0 +1,111 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Medication;
+import seedu.address.model.person.Patient;
+import seedu.address.model.person.Person;
+
+/**
+ * Changes the medication of an existing person in the address book.
+ */
+public class PrescribeCommand extends Command {
+
+    public static final String COMMAND_WORD = "prescribe";
+
+    //@@author Jeffry Lum-reused
+    //Reused from https://nus-cs2103-ay2223s2.github.io/tp/tutorials/AddRemark.html
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Changes current medication information of the person identified "
+            + "by the index number used in the last person listing. "
+            + "Existing medication information will be overwritten by input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "m/ [MEDICATION]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "m/ paracetamol";
+
+    public static final String MESSAGE_ADD_PRESCRIBE_SUCCESS = "Added medication to Person: %1$s";
+    public static final String MESSAGE_DELETE_PRESCRIBE_SUCCESS = "Removed medication from Person: %1$s";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_SELECTED_DOCTOR_FAILURE = "You have selected a doctor, this command is only "
+            + "applicable for doctors.";
```

Comment:
I like how you are handling possible errors and interactions with the users.  Perhaps we could change the phrasing of "medication" to "medication prescription" to better describe the final action. Also, I think 'MESSAGE_SELECTED_DOCTOR_FAILURE' should end with "applicable for patients."